amr-wind/equation_systems/icns/icns_advection.H

```diff
@@ -576,6 +563,24 @@ struct AdvectionOp<ICNS, fvm::MOL>
             for (amrex::MFIter mfi(dof_field(lev), mfi_info); mfi.isValid();
                  ++mfi) {
                 amrex::Box const& bx = mfi.tilebox();
+                amrex::Box gbx = grow(bx, fvm::MOL::nghost_state);
+
+                // Set up momentum array
+                amrex::FArrayBox qfab;
+                qfab.resize(gbx, ICNS::ndim);
+                amrex::Array4<amrex::Real> q;
```

Comment:
and do `const auto& q = qfab.array();` ?

Reply:
good catch. Added this change to a new commit as well. Will push soon.